webrender/src/renderer.rs

```diff
@@ -1131,7 +1117,22 @@ impl Renderer {
                                      options.precache_shaders)
         };
 
+        let device_max_size = device.max_texture_size();
+        // 512 is the minimum that the texture cache can work with.
+        // Broken GL contexts can return a max texture size of zero (See #1260). Better to
+        // gracefully fail now than panic as soon as a texture is allocated.
+        let min_texture_size = 512;
+        if device_max_size < min_texture_size {
+            println!("Device reporting insufficient max texture size ({})", device_max_size);
+            return Err(InitError::MaxTextureSize);
+        }
+        let mut max_texture_size = cmp::max(
+            cmp::min(device_max_size, options.max_texture_size.unwrap_or(device_max_size)),
+            min_texture_size
+        );
```

Comment:
Thanks, I move that block back.

The TextureCache is passed to render_backend. Maybe we can't always get a device when we try to call TextureCache::compute_max_texture_size().